### ISSUE TITLE
Framed mode changes vertical axis camera position when not supposed to if out of horizontal bounds. Godot 4.1

### ISSUE THREAD

**Issue by KayCat88** [ramokz/phantom-camera#144]
### Issue description

I have the vertical dead zone set to a size which makes it so when the character jumps, it doesn't move the camera. However, if the character is moving out of the horizontal bound and the camera is following it, the camera moves vertically when the player jumps, which doesn't happen if the player is still (due to the aforementioned vertical dead zone size). I've found that this also occurs with the horizontal axis. I don't know if this is intended to happen, but I assume not. My apologies if it is.
![image](https://github.com/ramokz/phantom-camera/assets/131679960/9754dc63-6c6a-4a8a-896e-8d9eaf53dd0f)

https://github.com/ramokz/phantom-camera/assets/131679960/95c8ae74-cab9-49b4-8470-8e8693e37ddd

### Steps to reproduce

To reproduce this, the only things necessary are the following.

- Have a character that can move along the screen in both axis (this could be done with the character movement script that comes with the CharacterBody2D node and a platform to move along).
- With the add-on in the project, set the character as the target.
- Set the follow mode to Framed and to set the dead zone values to something that doesn't move the camera in one axis (Y for example) with the character's movement.
- Get out of bounds with the other axis (X) and move the player in the axis that originally wasn't supposed to move the camera (Y).

### (Optional) Minimal reproduction project

[Framed Mode.zip](https://github.com/ramokz/phantom-camera/files/13536620/Framed.Mode.zip)

**Comment by ramokz** [comment]
Correct, it is not an intentional result of the feature, but rather a shortcoming that's still being figured out.

Another thing I spotted in your video example is that the target seems to have exceeded the bounds of the dead zone. Though I'm not quite sure how that occurred, since nothing unusual appeared to happen as you moved the character around?

**Comment by Sarrg** [comment]
I think this is the same issue as I was having right now.
The cause for this is that there is only a check if the target is touching a deadzone and disregards on which axis.
Thus the global position is updated to the `target_position`.
What need to be done is that global position horizontally (x position) is only updated when the horizontal dead zone is reached and the same for the vertical.

My quick fix for this is to add the following to `phantom_camera_2D.gd:523`:
```godot
if offset.x == 0:    # offset = Properties.get_framed_side_offset() 
    target_position.x = get_global_position().x
if offset.y == 0:
    target_position.y = get_global_position().y
```
and remove the `else` to always update `_camera_offset`.

A more proper fix would be to only store `_target_position_with_offset()` as `_camera_offset` and then do:
```godot
target_position = get_global_position() + abs(Properties.get_framed_side_offset()) * (_target_position_with_offset() - _camera_offset)
```

**Comment by ramokz** [comment]
Should be resolved in #373 in v0.8
